en/ros2/px4_ros2_interface_lib.md

```diff
@@ -0,0 +1,365 @@
+2. Clone the repository into the workspace:
+
+   ```sh
+   cd $ros_workspace/src
+   git clone --recursive https://github.com/Auterion/px4-ros2-interface-lib
+   ```
+
+   :::note
+   To ensure compatibility, use the latest _main_ branches for PX4, _px4_msgs_ and the library.
+   See also [here](https://github.com/Auterion/px4-ros2-interface-lib#compatibility-with-px4).
+   :::
+
+3. Build the workspace:
+
+   ```sh
+   cd ..
+   source install/setup.bash
+   colcon build
+   ```
+
+4. In a different shell, start PX4 SITL:
+
+   ```sh
+   cd $px4-autopilot
+   make px4_sitl gazebo-classic
+   ```
+
+   (here we use Gazebo-Classic, but you can use any model or simulator)
+
+5. Run the micro XRCE agent in a new shell (you can keep it running afterward):
+
+   ```sh
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+6. Start QGroundControl.
+
+   :::note
+   Use QGroundControl Daily, which supports dynamically updating the list of modes.
+   :::
+
+7. Back in the ROS 2 terminal, run one of the example modes:
+
+   ```shell
+   ros2 run example_mode_manual_cpp example_mode_manual
+   ```
+
+   You should get an output like this showing 'My Manual Mode' mode being registered:
+
+   ```sh
+   [DEBUG] [example_mode_manual]: Checking message compatibility...
+   [DEBUG] [example_mode_manual]: Subscriber found, continuing
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Registering 'My Manual Mode' (arming check: 1, mode: 1, mode executor: 0)
+   [DEBUG] [example_mode_manual]: Subscriber found, continuing
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Got RegisterExtComponentReply
+   [DEBUG] [example_mode_manual]: Arming check request (id=1, only printed once)
+   ```
+
+8. On the PX4 shell, you can check that PX4 registered the new mode:
+
+   ```sh
+   commander status
+   ```
+
+   The output should contain:
+
+   ```{5}
+   INFO  [commander] Disarmed
+   INFO  [commander] navigation mode: Position
+   INFO  [commander] user intended navigation mode: Position
+   INFO  [commander] in failsafe: no
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+   ```
+
+9. At this point you should be able to see the mode in QGroundControl as well:
+
+   ![QGC Modes](../../assets/middleware/ros2/px4_ros2_interface_lib/qgc_modes.png)
+
+10. Select the mode, make sure you have a manual control source (physical or virtual joystick), and arm the vehicle.
+    The mode will then activate, and it should print the following output:
+
+    ```sh
+    [DEBUG] [example_mode_manual]: Mode 'My Manual Mode' activated
+    ```
+
+11. Now you are ready to create your own mode.
+
+## How to Use the Library
+
+The following sections describe specific functionality provided by this library.
+In addition, any other PX4 topic can be subscribed or published.
+
+### Mode Class Definition
```

Comment:
We have no Mode Executor definition. Is that planned or just delayed, or what? It would be good to provide a realistic example.

Reply:
I can add that.